src/pages/index.js

```diff
@@ -1,129 +1,239 @@
-  return newCard.generateCard();
+function createCard(data) {
+  const card = new Card(
+    data,
+    userInfo.getUserId(),
+    "#card-template",
+    handleImageClick,
+    (cardId, cardElement) => {
+      return new Promise((resolve, reject) => {
+        confirmPopup.setSubmitAction(() => {
+          confirmPopup.renderLoading(true);
+          api
+            .deleteCard(cardId)
+            .then(() => {
+              cardElement.remove();
+              resolve();
+              confirmPopup.close();
+            })
+            .catch((err) => {
+              showError(err);
+              reject(err);
+            })
+            .finally(() => confirmPopup.renderLoading(false));
+        });
+        confirmPopup.open();
+      });
+    },
+    handleLikeClick
+  );
+  return card.generateCard();
 }
 
-openAddpopupButton.addEventListener("click", () => {
-  console.log("Add button clicked");
-  formValidators["add-form"].resetValidation(); 
-  addCardPopup.open();
-});
+function renderCard(data) {
+  const cardElement = createCard(data);
+  cardSection.addItem(cardElement);
+}
 
-document.addEventListener("DOMContentLoaded", () => {
-  cardSection.renderItems();
-});
+function handleImageClick(name, link) {
+  imagePopup.open({ name, link });
+}
 
-window.addEventListener("load", () => {});
\ No newline at end of file
+function handleProfileFormSubmit(formData) {
+  editProfilePopup.renderLoading(true);
+  api
+    .updateUserInfo(formData)
+    .then((res) => {
+      userInfo.setUserInfo({ name: res.name, about: res.about });
+      editProfilePopup.close();
+    })
+    .catch((err) => showError(err))
+    .finally(() => editProfilePopup.renderLoading(false));
+}
+
+function handleAddCardFormSubmit(formData) {
+  addCardPopup.renderLoading(true);
+  api
+    .addNewCard(formData)
+    .then((cardData) => {
+      renderCard(cardData);
+      addCardPopup.close();
+    })
+    .catch((err) => showError(err))
+    .finally(() => addCardPopup.renderLoading(false));
+}
+
+function handleAvatarFormSubmit(formData) {
+  avatarPopup.renderLoading(true);
+  api
+    .updateAvatar({ avatar: formData.avatar })
+    .then((res) => {
+      userInfo.setUserInfo({ avatar: res.avatar });
+      avatarPopup.close();
+    })
+    .catch((err) => showError(err))
+    .finally(() => avatarPopup.renderLoading(false));
+}
+
+function handleLikeClick(cardId, isLiked, cardInstance) {
+  const action = isLiked ? api.removeLike(cardId) : api.addLike(cardId);
+  action
+    .then((updatedCard) => {
+      console.log("API Like Response:", updatedCard);
+      let likes = Array.isArray(updatedCard.likes) ? updatedCard.likes : [];
+      if (!likes.length) {
+        console.log("Fallback: manually updating likes...");
+        const currentUserId = userInfo.getUserId();
+        likes = isLiked
+          ? cardInstance._likes.filter((user) => user._id !== currentUserId)
+          : [...cardInstance._likes, { _id: currentUserId }];
```

Comment:
* `likes` don't exist inside the card data. There is `isLiked` for checking likes

Please, don't check them. You have `isLiked` inside the `updatedCard`